datasync/chaindatafetcher/kafka/consumer_test.go

```diff
@@ -556,3 +558,157 @@ func Test_V1Segment_MultiProducers(t *testing.T) {
 		}
 	}
 }
+
+func TestConsumer_handleError(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	testSegment := &Segment{
+		orig: &sarama.ConsumerMessage{
+			Topic:     "test-topic",
+			Partition: int32(1),
+			Offset:    int64(99),
+		},
+	}
+
+	testSegment2 := &Segment{
+		orig: &sarama.ConsumerMessage{
+			Topic:     "test-topic",
+			Partition: int32(1),
+			Offset:    int64(100),
+		},
+	}
+
+	session := mocks.NewMockConsumerGroupSession(ctrl)
+	parentErr := errors.New("parent error")
+	callbackErr := errors.New("callback error")
+
+	tests := []struct {
+		name      string
+		setup     func()
+		callback  func(s string) error
+		buffer    [][]*Segment
+		parentErr error
+		expected  error
+	}{
+		{
+			"no_item_in_buffer",
+			func() {},
+			func(s string) error { panic("should not be called") },
+			[][]*Segment{},
+			parentErr,
+			parentErr,
+		},
+		{
+			"no_callback",
+			func() {},
+			nil,
+			[][]*Segment{{testSegment}},
+			parentErr,
+			parentErr,
+		},
+		{
+			"callback_success_an_item_in_buffer",
+			func() { session.EXPECT().MarkMessage(gomock.Eq(testSegment.orig), gomock.Eq("")).Times(1) },
+			func(s string) error { return nil },
+			[][]*Segment{{testSegment}},
+			parentErr,
+			nil,
+		},
+		{
+			"callback_success_two_items_in_buffer",
```

Comment:
Can I ask why you add this tc? because i think `callback_success_two_items_in_buffer` tc and `callback_failure` tc have same purpose..

Reply:
@yoomee1313 Oh, `callback_success_two_items_in_buffer` is a successful case by calling error callback. On the other hand, the `callback_failure` is a failure case although the error callback is called.

Reply:
Oh, setup is different.. thanks for the explanation!